native/src/main/java/io/ballerina/lib/np/Native.java

```diff
@@ -0,0 +1,195 @@
+        bStringValues[1] = StringUtils.fromString("array");
+        bStringValues[2] = StringUtils.fromString("string");
+        bStringValues[3] = StringUtils.fromString("number");
+        bStringValues[4] = StringUtils.fromString("boolean");
+        bStringValues[5] = StringUtils.fromString("null");
+        BMap<BString, Object> schemaMap = createMapValue(TypeCreator.createMapType(PredefinedTypes.TYPE_JSON));
+        schemaMap.put(StringUtils.fromString("type"), ValueCreator.createArrayValue(bStringValues));
+        return schemaMap;
+    }
+
+    private static Object generateJsonSchemaForArrayType(ArrayType arrayType,
+                                                         SchemaGenerationContext schemaGenerationContext) {
+        BMap<BString, Object> schemaMap = createMapValue(TypeCreator.createMapType(PredefinedTypes.TYPE_JSON));
+        Type elementType = TypeUtils.getReferredType(arrayType.getElementType());
+        schemaMap.put(StringUtils.fromString("type"), StringUtils.fromString("array"));
+        schemaMap.put(StringUtils.fromString("items"), generateJsonSchemaForType(elementType,
+                                                                                    schemaGenerationContext));
+        return schemaMap;
+    }
+
+    private static Object generateJsonSchemaForTupleType(TupleType tupleType,
+                                                         SchemaGenerationContext schemaGenerationContext) {
+        BMap<BString, Object> schemaMap = createMapValue(TypeCreator.createMapType(PredefinedTypes.TYPE_JSON));
+        schemaMap.put(StringUtils.fromString("type"), StringUtils.fromString("array"));
+        BArray annotationArray = ValueCreator.createArrayValue(TypeCreator.createArrayType(PredefinedTypes.TYPE_JSON));
+        int index = 0;
+        for (Type type : tupleType.getTupleTypes()) {
+            annotationArray.add(index++, generateJsonSchemaForType(type, schemaGenerationContext));
+        }
+        schemaMap.put(StringUtils.fromString("items"), annotationArray);
+        return schemaMap;
+    }
+
+    private static boolean isSimpleType(Type type) {
+        return type.getBasicType().all() <= 0b100000;
+    }
+
+    private static String getStringRepresentation(Type type) {
+        return switch (type.getBasicType().all()) {
+            case 0b000000 -> "null";
+            case 0b000010 -> "boolean";
+            case 0b000100 -> "integer";
+            case 0b001000, 0b010000 -> "number";
+            case 0b100000 -> "string";
+            default -> null;
+        };
+    }
+
+    private static Object generateJsonSchemaForRecordType(RecordType recordType,
+                                                          SchemaGenerationContext schemaGenerationContext) {
+        for (Map.Entry<BString, Object> entry : recordType.getAnnotations().entrySet()) {
+            if ("ballerina/np:0:Schema".equals(entry.getKey().getValue())) {
```

Comment:
It is better if we can save the version as a constant.
may be we can do it in another PR

Reply:
Yeah, this just moves the existing code.